torch/distributed/_composable/replicate.py

```diff
@@ -4,7 +4,30 @@
 import torch.nn as nn
 
 from . import _ddp
-from .contract import contract
+from .contract import contract, _get_registry
+
+
+@contract()
+def replicate(
+    module: nn.Module,  # NOTE: contract now supports single module only
+    **kwargs,
+) -> nn.Module:
+    r"""Replicates a module
+
+    Args:
+        module (torch.nn.Module): module to replicate
+
+    Example::
+        >>> module = nn.Linear(3, 3)
+        >>> replicate(module)
+    """
```

Comment:
can be done in followup PRs, but we will need a much more comprehensive API docstring, e.g., what does this do, what are expected kwargs, how does this compose with others, and what are mutual-exclusive APIs with this one, etc.